bin/reth/src/main.rs

```diff
@@ -4,9 +4,6 @@
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
 fn main() {
-    // ensure we have the correct features enabled
-    reth::primitives::ensure_ethereum!();
-
```

Comment:
we still want this, but now in bin/reth and for the optimism feature instead

Reply:
Added a runtime check in the node startup. In the future, we can split out into separate binaries for release safety and further separation 😄